src/toil/lib/misc.py

```diff
@@ -123,16 +123,16 @@ def atomic_copy(src_path, dest_path, executable=False):
     """Copy a file using posix atomic creations semantics."""
     with AtomicFileCreate(dest_path) as dest_path_tmp:
         shutil.copyfile(src_path, dest_path_tmp)
-    if executable:
-        os.chmod(dest_path, os.stat(dest_path).st_mode | stat.S_IXUSR)
+        if executable:
+            os.chmod(dest_path, os.stat(dest_path).st_mode | stat.S_IXUSR)
```

Comment:
This has to operate on `dest_path_tmp` here, not `dest_path`. When you leave the `with` block, any file at `dest_path` (there might not have been one) is replaced with the one at `dest_path_tmp`.